src/components/expanded-state/ChartExpandedState.js

```diff
@@ -201,29 +205,35 @@ export default function ChartExpandedState({ asset }) {
           )}
         </TokenInfoRow>
       </TokenInfoSection>
-      <ActionRow key={`row${showChart}`}>
-        {showSwapButton && (
-          <SwapActionButton
+      {needsEth ? (
+        <ActionRow key={`row${showChart}`}>
+          <BuyActionButton color={color} />
+        </ActionRow>
+      ) : (
+        <ActionRow key={`row${showChart}`}>
+          {showSwapButton && (
+            <SwapActionButton
+              color={color}
+              inputType={AssetInputTypes.in}
+              radiusAndroid={24}
+              radiusWrapperStyle={{ flex: 1, marginRight: 10 }}
+              wrapperProps={{
+                containerStyle: { flex: 1 },
+                style: { flex: 1 },
+              }}
+            />
+          )}
+          <SendActionButton
             color={color}
-            inputType={AssetInputTypes.in}
             radiusAndroid={24}
-            radiusWrapperStyle={{ flex: 1, marginRight: 10 }}
+            radiusWrapperStyle={{ flex: 1, marginLeft: 10 }}
```

Comment:
I've changed this so many times the past week that I'm forced to ask I if it's really needed and tested on both OS.